server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -1522,11 +1522,24 @@ public void submitTask(String source, T task, @Nullable TimeValue timeout) {
             final var taskHolder = new AtomicReference<>(task);
             final Scheduler.Cancellable timeoutCancellable;
             if (timeout != null && timeout.millis() > 0) {
-                timeoutCancellable = threadPool.schedule(
-                    new TaskTimeoutHandler<>(timeout, source, taskHolder),
-                    timeout,
-                    threadPool.generic()
-                );
+                try {
+                    timeoutCancellable = threadPool.schedule(
+                        new TaskTimeoutHandler<>(timeout, source, taskHolder),
+                        timeout,
+                        threadPool.generic()
+                    );
+                } catch (Exception e) {
```

Comment:
Should it be more specific and only catch rejection?

Reply:
I think it can only be a rejection, but there's no good reason to let anything else escape here. I added an assertion in 01e4c816cf3.